activerecord/test/cases/finder_test.rb

```diff
@@ -1883,6 +1883,15 @@ def test_finder_with_offset_string
     end
   end
 
+  test "#reverse_order for a model with composite query constraints" do
+    quoted_type = Regexp.escape(quote_table_name("clothing_items.clothing_type"))
+    quoted_color = Regexp.escape(quote_table_name("clothing_items.color"))
+
+    assert_queries_match(/ORDER BY #{quoted_type} DESC, #{quoted_color} DESC LIMIT/i) do
```

Comment:
Without patch:
![Screenshot 2025-02-24 at 8 15 16 pm](https://github.com/user-attachments/assets/b493f251-2cfd-412e-ac8c-0c19fea17757)

The expected query is the same as `ClothingItem.last` tested above.

Reply:
Rather than to assert on the exact query, wouldn't it be preferable to assert on the records order?

e.g.

```ruby
assert_equal ClothingItem.pluck(:id).reverse, ClothingItem.reverse_order.pluck(:id)
```

Reply:
Can do. Actually this test isn't appropriate here cause `#reverse_order` is not a finder method... Will address.

Reply:
Addressed 👍🏽